reviewing/license-scanner/src/getPkgInfo.ts

```diff
@@ -259,7 +259,7 @@ export async function readPackageIndexFile (
     pkgIndexFilePath = getIndexFilePathInCafs(
       opts.storeDir,
       packageResolution.integrity as string,
-      `${parsedId.name}@${parsedId.version}`
+      parsedId.nonSemverVersion ?? `${parsedId.name}@${parsedId.version}`
```

Comment:
are you sure this is the right fix? It could be also
```suggestion
      `${parsedId.name}@${parsedId.nonSemverVersion ?? parsedId.version}`
```

Reply:
Looking at `resolveDependencies.ts`, [nonSemverVersion is used when it is present](https://github.com/pnpm/pnpm/blob/b0ead519b3338f769619029e542233bea3266cf9/pkg-manager/resolve-dependencies/src/resolveDependencies.ts#L1182C14-L1182C30) and the `${parsedId.name}@${parsedId.version}` format is only used when valid semver is present

Reply:
@zkochan I've tested this and confirmed this is the correct fix. Is there any chance I can get another look at this PR?